src/GafferSceneUI/LightTool.cpp

```diff
@@ -704,52 +707,81 @@ class LightToolHandle : public Handle
 
 	public :
 
-		LightToolHandle(
-			const std::string &lightTypePattern,
-			const std::string &name
-		) :
-			Handle( name ),
-			m_lightTypePattern( lightTypePattern ),
-			m_lookThroughLight( false )
+		struct InspectionInfo
 		{
+			Inspector::ResultPtr inspection;
+			/// \todo Template the value so it can be other things than `float`?
+			float value;
+		};
+
+		using InspectionMap = std::unordered_map<InternedString, InspectionInfo>;
 
-		}
 		~LightToolHandle() override
 		{
 
 		}
 
 		// Update inspectors and data needed to display and interact with the tool. Called
 		// in `preRender()` if the inspections are dirty.
-		// Derived classes should call this parent method first, then implement custom logic.
-		virtual void update( ScenePathPtr scenePath, const PlugPtr &editScope )
+		void setScenePath( ScenePathPtr scenePath )
 		{
 			m_handleScenePath = scenePath;
-			m_editScope = editScope;
-		}
 
-		bool isLightType( const std::string &shaderAttribute ) const
-		{
-			auto lightType = Metadata::value<StringData>( shaderAttribute, "type" );
+			m_inspectors.clear();
 
-			if( !lightType || !StringAlgo::matchMultiple( lightType->readable(), m_lightTypePattern ) )
+			if( !getScenePath()->isValid() )
 			{
-				return false;
+				return;
 			}
 
-			return true;
-		}
+			m_editScope = m_view->editScopePlug();
 
-		ScenePath *handleScenePath() const
-		{
-			return m_handleScenePath.get();
+			/// \todo This can be simplified and some of the logic, especially getting the inspectors, can
+			/// be moved to the constructor when we standardize on a single USDLux light representation.
+
+			ConstCompoundObjectPtr attributes = getScenePath()->getScene()->fullAttributes( getScenePath()->names() );
+
+			for( const auto &[attributeName, value ] : attributes->members() )
+			{
+				if(
+					StringAlgo::match( attributeName, g_lightAttributePattern ) &&
+					value->typeId() == (IECore::TypeId)ShaderNetworkTypeId
+				)
+				{
+					const auto shader = attributes->member<ShaderNetwork>( attributeName )->outputShader();
+					std::string shaderAttribute = shader->getType() + ":" + shader->getName();
+
+					if( !isLightType( shaderAttribute ) )
+					{
+						continue;
+					}
+
+					for( const auto &m : m_metaParameters )
+					{
+						if( auto parameter = Metadata::value<StringData>( shaderAttribute, m ) )
+						{
+							m_inspectors[m] = new ParameterInspector(
+								getScenePath()->getScene(),
+								m_editScope,
+								attributeName,
+								ShaderNetwork::Parameter( "", parameter->readable() )
+							);
+						}
+					}
+
+					break;
+				}
+			}
+
+			setScenePathInternal( scenePath );
 		}
 
-		Plug *editScope() const
+		ScenePath *getScenePath() const
```

Comment:
Should this be returning `const ScenePath *`, so nobody can mess with it? I wonder if `get/setHandlePath()` would be a better name - that seems to add a bit more information about the purpose of the path without being any longer. The fact that the corresponding member data includes "handle" suggests that it might have been a useful distinction to you already?

Reply:
In https://github.com/GafferHQ/gaffer/pull/5483/commits/82e367704cc99ce474e7a68799b07d313d3ccfba I did a broader reworking that I think is necessary. Here's my reasoning :

Returning `const ScenePath *` does sound good, but the return value of that method was being used to construct a `ParameterInspector`. `ParameterInspector` wants a `const` reference to the `ScenePlugPtr` which can't be implicitly created from a `const ScenePath *

I'm not actually sure why this was both compiling and not crashing before.

I think a better solution is in https://github.com/GafferHQ/gaffer/pull/5483/commits/82e367704cc99ce474e7a68799b07d313d3ccfba where I have `LightToolHandle` own the pointer to the `ScenePlug` in `m_scene`, hold a non-owning pointer the context and hold the `ScenePlug::PathPlug`. All separately rather than in a `ScenePath` object.

I think that's the most correct way to handle the `ScenePlug` and keep `ParameterInspector` working right. It also has a side benefit of being a bit more direct with the context and handle path, rather than getting it from a `ScenePath` every time.

I also changed the name to `updateHandlePath()`.  I'm not 100% sold on that name because it kind of wraps up a `ScenePlug`, `Context` and `ScenePlug::Path` all into the concept of a "path", but maybe the precedent of `ScenePath` class justifies that?